hypothesis-python/src/hypothesis/core.py

```diff
@@ -926,11 +929,11 @@ def wrapped_test(*arguments, **kwargs):
                             'You can add @seed(%(seed)d) to this test or run '
                             'pytest with --hypothesis-seed=%(seed)d to '
                             'reproduce this failure.') % {
-                                'seed': generated_seed},)
+                                'seed': generated_seed}, settings)
                     else:
                         report((
                             'You can add @seed(%d) to this test to reproduce '
-                            'this failure.') % (generated_seed,))
+                            'this failure.') % (generated_seed,), settings)
```

Comment:
Similarly, these calls could potentially use `with local_settings(settings):` instead of passing the setting manually.

There's some similar-looking code in `engine.py` that already does:

```python
    def debug(self, message):
        with local_settings(self.settings):
            debug_report(message)
```